app/grandchallenge/reader_studies/tasks.py

```diff
@@ -124,3 +124,15 @@ def add_image_to_answer(*, upload_session_pk, answer_pk):
         add_image(answer, image)
     else:
         raise ValueError("Upload session for answer does not match")
+
+
+@shared_task(**settings.CELERY_TASK_DECORATOR_KWARGS["acks-late-2xlarge"])
+def copy_reader_study_display_sets(*, orig_pk, new_pk):
+    orig = ReaderStudy.objects.get(pk=orig_pk)
+    new = ReaderStudy.objects.get(pk=new_pk)
+    if orig.display_sets.count() == new.display_sets.count():
```

Comment:
Why is this check needed?

Reply:
That should only happen in development though. If you did that in production this check wouldn't help as the two tasks would likely run in parallel. If the task is only called once then the atomic block is enough protection as SQS and celery with acks late guarantees only one execution of the task, but it could get called many times before completion which is where the atomic block adds its protection.

Reply:
Ah thank you for explaining! I was not sure if the atomic block was enough when I accidentally double called it, but that makes a lot of sense. I'll remove the check.